sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1240,6 +1240,40 @@ class Dataset[T] private[sql](
     joinWith(other, condition, "inner")
   }
 
+  /**
+   * Joins this Dataset returning value of left where `condition` evaluates to true.
+   *
+   * This is similar to the relation `join` function with one important difference in the
+   * result schema. Since `joinPartial` preserves objects present on left side of the join, the
+   * result schema is similarly nested into one column names `_1`.
+   *
+   * This type of join can be useful both for preserving type-safety with the original object
+   * types as well as working with relational data where either side of the join has column
+   * names in common.
+   *
+   * @param other Right side of the join.
+   * @param condition Join expression.
+   * @param joinType Type of join to perform. Default `inner`. Must be one of:
```

Comment:
Should we rephrase this comment since inner join is not supported? Like the default type can be probably `LeftSemi` or `LeftAnti` since `Inner` is not supported.

Reply:
> Should we rephrase this comment since inner join is not supported? Like the default type can be probably `LeftSemi` or `LeftAnti` since `Inner` is not supported.

Sorry, forgot remove the `Default inner`